httpclient/HTTPClient.cpp

```diff
@@ -19,6 +19,24 @@ long		HTTPClient::m_iConnectionTimeout = 10;
 long		HTTPClient::m_iTimeout = 90; //max, time that a download has to be finished?
 std::string	HTTPClient::m_sUserAgent = "domoticz/1.0";
 
+size_t write_curl_headerdata(void *contents, size_t size, size_t nmemb, void *userp)
+{
```

Comment:
I think you could use the 'StringSplit' function to quickly get the vector filled ? (\n\r as delimiter)

Reply:
Why do you need the returned headers? Make it a VERY valid reason because the code gets dirty now.
If possible, revert this change!

Reply:
Thats for the headsup! Its clear, but i dont think you need the headers for this.
Next, this still might crash, image you do not have the credentials to access this url and you get a json back with an error message, then 'item.json.sensors['outside']' does not exists and needs to be checked.
A simple function IsJSon or IsXML could also be created.
Calling OpenURL could also just return true/false, if there is an error (status != 200) there is no need to continue parsing anyway.
When you run Domoticz in debug of have the trace enabled you already see the error/status code printed in the log.
Its just that i want to keep the HTTPClient class as clean as possible.
Also the new way of parsing the status code for logging is not great. It makes the code cluttered and additional code has to be created

Reply:
If you do not have the permission you will get a 401 as a statuscode and item.isOk or item.headers.statusCode != 200. Of course you have to check that before you access the response data in your script. But even if you get a 401 there is a mime-type and only if that is json then dzVents will do the parsing (as a convenience to the user). 
So, as far as I am concerned I'm fine with handing over just the statuscode and the mime-type in the callback instead of all the headers. But I do need those two as the minimum.

Reply:
how about

```
if (domoticz.openURL(...)) then
 ... all is fine
end
```

Reply:
About StringSplit; I just copy/pasted and changed the already used callback function  write_curl_data_single_line which is used below. Didn't look into the possibility to use stringsplit for this, might be an option.

Reply:
It is asynchronous so you cannot wait for the return value. OpenURL only fills the commandArray and long after the script is finished the results will be fetched by Domoticz and this will again trigger dzVents. With the callback trigger dzVents will see which script needs to be called with the response data.